src/Microsoft.DotNet.Tools.Builder/CompileContext.cs

```diff
@@ -338,23 +338,26 @@ public CompilerIO(List<string> inputs, List<string> outputs)
         public static CompilerIO GetCompileIO(ProjectContext project, string config, string outputPath, string intermediaryOutputPath, ProjectDependenciesFacade dependencies)
         {
             var compilerIO = new CompilerIO(new List<string>(), new List<string>());
+            var compilationOutput = CompilerUtil.GetCompilationOutput(project.ProjectFile, project.TargetFramework, config, outputPath);
 
             // input: project.json
             compilerIO.Inputs.Add(project.ProjectFile.ProjectFilePath);
 
+            // input: lock file; find when dependencies change
+            compilerIO.Inputs.Add(project.LockFile.LockFilePath);
```

Comment:
It can't be a restore error (since the user may not have run restore yet and thus has no lock file), it *is* a compile error however. Compile handles this but I'm not sure build does. Adding it test can't hurt and it'll be a common scenario.

https://github.com/dotnet/cli/blob/master/src/Microsoft.DotNet.ProjectModel/ProjectContextBuilder.cs#L205